dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CollectionRestRepository.java

```diff
@@ -253,4 +262,28 @@ protected void delete(Context context, UUID id) throws AuthorizeException {
             throw new RuntimeException("Unable to delete collection because the logo couldn't be deleted", e);
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * Method to install a logo on a Collection which doesn't have a logo
+     * Called by request mappings in CollectionRestController
+     * @param context
+     * @param collection    The collection on which to install the logo
+     * @param uploadfile    The new logo
+     * @return              The created bitstream containing the new logo
+     * @throws IOException
+     * @throws AuthorizeException
+     * @throws SQLException
+     */
+    public Bitstream setLogo(Context context, Collection collection, MultipartFile uploadfile)
```

Comment:
RestRepositories generally don't return JPA entity objects. They return RestModels. So in this example, it'd be BitstreamRest that would be returned.

Reply:
Great catch @cwilper, I've updated this accordingly to match the other RestRepository methods